lib/wallets/wallet/wallet_mixin_interfaces/electrumx_interface.dart

```diff
@@ -793,10 +803,78 @@ mixin ElectrumXInterface<T extends Bip39HDCurrency> on Bip39HDWallet<T> {
   }
 
   Future<int> fetchChainHeight() async {
+    final Completer<int> completer = Completer<int>();
+
     try {
-      final result = await electrumXClient.getBlockHeadTip();
-      return result["height"] as int;
-    } catch (e) {
+      // Don't set a stream subscription if one already exists.
+      if (ElectrumxChainHeightService.subscriptions[cryptoCurrency.coin] !=
```

Comment:
This does not guarantee that the subscription will not be set again as we don't know for sure whether the _latestHeight was set so this may not return early

Reply:
3d42967c8b5f1a1ae515956b8b80ee87f3f7a444 should address this issue, setting the subscription via null assignment

Reply:
46285d44ea0fe1ce8ed83f2930b523e874b3f7d9 should address this